services/f-droid/f-droid.tester.js

```diff
@@ -8,165 +8,77 @@ const t = (module.exports = new ServiceTester({
+  .intercept(nock => nock(base).get(path).reply(200, testJson))
+  .expectBadge({ label: 'f-droid', message: 'v0.2.7' })
+
+t.create('Package is not found with 403')
+  .get('/v/org.fdroid.fdroid.privileged.json')
+  .intercept(nock => nock(base).get(path).reply(403, 'some 403 text'))
+  .expectBadge({ label: 'f-droid', message: 'app not found' })
 
-t.create('Package is not found with "metadata_format" query parameter')
-  .get('/v/axp.tool.apkextractor.json?metadata_format=yml')
-  .intercept(nock => nock(base).get(`${path}.yml`).reply(404))
+t.create('Package is not found with 404')
+  .get('/v/org.fdroid.fdroid.privileged.json')
+  .intercept(nock => nock(base).get(path).reply(404, 'some 403 text'))
   .expectBadge({ label: 'f-droid', message: 'app not found' })
 
-t.create('Package is found yml matadata format with missing "CurrentVersion"')
-  .get('/v/axp.tool.apkextractor.json?metadata_format=yml')
+t.create('Package is not found with no packages available (empty array)"')
+  .get('/v/org.fdroid.fdroid.privileged.json')
   .intercept(nock =>
-    nock(base).get(`${path}.yml`).reply(200, 'Categories: System')
+    nock(base)
+      .get(path)
+      .reply(
+        200,
+        '{"packageName":"org.fdroid.fdroid.privileged","packages":[]}'
+      )
   )
-  .expectBadge({ label: 'f-droid', message: 'invalid response data' })
+  .expectBadge({ label: 'f-droid', message: 'app not found' })
 
-t.create('Package is found with bad yml matadata format')
-  .get('/v/axp.tool.apkextractor.json?metadata_format=yml')
+t.create('Package is not found with no packages available (missing array)"')
+  .get('/v/org.fdroid.fdroid.privileged.json')
   .intercept(nock =>
-    nock(base).get(`${path}.yml`).reply(200, '.CurrentVersion: 1.4')
+    nock(base)
+      .get(path)
+      .reply(200, '{"packageName":"org.fdroid.fdroid.privileged"}')
   )
-  .expectBadge({ label: 'f-droid', message: 'invalid response data' })
-
-t.create('Package is not found')
-  .get('/v/axp.tool.apkextractor.json')
-  .intercept(nock => nock(base).get(`${path}.txt`).reply(404))
-  .intercept(nock => nock(base).get(`${path}.yml`).reply(404))
   .expectBadge({ label: 'f-droid', message: 'app not found' })
 
-t.create('The api changed')
-  .get('/v/axp.tool.apkextractor.json?metadata_format=yml')
-  .intercept(nock => nock(base).get(`${path}.yml`).reply(200, ''))
+t.create('Package is not found with invalid response"')
+  .get('/v/org.fdroid.fdroid.privileged.json')
+  .intercept(nock => nock(base).get(path).reply(200, '{"field":"value"}'))
   .expectBadge({ label: 'f-droid', message: 'invalid response data' })
 
-t.create('Package is not found due invalid metadata format')
-  .get('/v/axp.tool.apkextractor.json?metadata_format=xml')
-  .expectBadge({
-    label: 'f-droid',
-    message: 'invalid query parameter: metadata_format',
-  })
-
 /* If this test fails, either the API has changed or the app was deleted. */
 t.create('The real api did not change')
```

Comment:
This test has a `timeout` of 10 seconds. Can it be removed, i.e. is the new API fast enough?

Reply:
For now it's moved from gitlab to gitlab. Later it will move to f-droid.org, which is actually even slower than gitlab. I actually don't really see timeout is used for other tests, so I can remove it, I just kept is there because it already was set.

Reply:
If the tests consistently pass without the timeout, I suggest removing it.

Reply:
For now it passes. When API will be moved to f-droid.org, it may be needed to bring it back. f-droid.org it known to be slow sometimes.